apps/cowswap-frontend/src/common/pure/AddressInputPanel/index.tsx

```diff
@@ -6,142 +6,216 @@ import {
   parsePrefixedAddress,
 } from '@cowprotocol/common-utils'
-import { useENS } from '@cowprotocol/ens'
-import { ExternalLink, RowBetween, UI } from '@cowprotocol/ui'
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+import { RowBetween, NetworkLogo } from '@cowprotocol/ui'
 import { useWalletInfo } from '@cowprotocol/wallet'
 
 import { t, Trans } from '@lingui/macro'
-import styled from 'styled-components/macro'
 
 import { AutoColumn } from 'legacy/components/Column'
 import { useIsDarkMode } from 'legacy/state/user/hooks'
 
+import { useEnhancedENS } from './hooks/useEnhancedENS'
+import {
+  InputPanel,
+  ContainerRow,
+  InputContainer,
+  Input,
+  DestinationChainInfo,
+  ValidationError,
+  StyledExplorerLink,
+} from './styled'
+
 import { autofocus } from '../../utils/autofocus'
 import ChainPrefixWarning from '../ChainPrefixWarning'
 
-const InputPanel = styled.div`
-  ${({ theme }) => theme.flexColumnNoWrap}
-  position: relative;
-  border-radius: 16px;
-  background-color: var(${UI.COLOR_PAPER_DARKER});
-  color: inherit;
-  z-index: 1;
-  width: 100%;
-`
-
-const ContainerRow = styled.div<{ error: boolean }>`
-  display: flex;
-  justify-content: center;
-  align-items: center;
-  border-radius: 16px;
-  border: 0;
-  color: inherit;
-  background-color: var(${UI.COLOR_PAPER_DARKER});
-`
-
-export const InputContainer = styled.div`
-  flex: 1;
-  padding: 1rem;
-`
-
-const Input = styled.input<{ error?: boolean }>`
-  font-size: 1.25rem;
-  outline: none;
-  border: none;
-  flex: 1 1 auto;
-  background: none;
-  transition: color 0.2s ${({ error }) => (error ? 'step-end' : 'step-start')};
-  color: ${({ error }) => (error ? `var(${UI.COLOR_DANGER})` : 'inherit')};
-  overflow: hidden;
-  text-overflow: ellipsis;
-  font-weight: 500;
-  width: 100%;
-
-  &&::placeholder {
-    color: inherit;
-    opacity: 0.5;
-  }
+function useInputHandler(
+  onChange: (value: string) => void,
+  addressPrefix: string | undefined,
+  setChainPrefixWarning: (warning: string) => void,
+): (event: ChangeEvent<HTMLInputElement>) => void {
+  return useCallback(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      const input = event.target.value
+      setChainPrefixWarning('')
+      let value = input.replace(/\s+/g, '')
 
-  &:focus::placeholder {
-    color: transparent;
-  }
+      if (isPrefixedAddress(value)) {
+        const { prefix, address } = parsePrefixedAddress(value)
 
-  padding: 0px;
-  appearance: textfield;
-  -webkit-appearance: textfield;
+        if (prefix && addressPrefix !== prefix) {
+          setChainPrefixWarning(prefix)
+        }
 
-  ::-webkit-search-decoration {
-    -webkit-appearance: none;
-  }
+        if (address) {
+          value = address
+        }
+      }
 
-  ::-webkit-outer-spin-button,
-  ::-webkit-inner-spin-button {
-    -webkit-appearance: none;
+      onChange(value)
+    },
+    [onChange, addressPrefix, setChainPrefixWarning],
+  )
+}
+
+function useCustomValidation(
```

Comment:
This file is supposed to be `AddressInputPanel` component file, but now it contains other components and hooks.
It's ok to keep 1-2 hooks or small components along with main one.
But here many of them and it worth it to separate hooks and other components from here.